components/ScheduleSection.js

```diff
@@ -0,0 +1,290 @@
+          />
+        </SessionRow>
+      </div>
+      <style jsx>{`
+        .schedule {
+          text-align: center;
+        }
+        .container {
+          max-width: 1060px;
+          width: 100%;
+          margin: 0 auto;
+          text-align: left;
+        }
+        @media (max-width: 768px){
+          .container {
+            max-width: 750px;
+            text-align: center;
+          }
+        }
+      `}</style>
+    </div>
+  )
+}
+
+const getSpeakerAvatar = (name) => {
+  return `static/speakers/${name.replace(' ', '')}.jpg`
+}
+
+const ActivityRow = ({ time, children }) => (
+  <div className='row _margin-bottom-30'>
+    <div className='column-md-2 column-xs-12 time'>{time}</div>
+    <div className='column-md-5 column-xs-12 main'>{children}</div>
+    <style jsx>{`
+      .row {
+        position: relative;
+        display: flex;
+        flex-wrap: wrap;
+        padding-left:7.5px;
+        padding-right:7.5px;
+      }
+      .column-md-5 {
+        width: 41.65%;
+      }
+      .column-md-2 {
+        width: 16.66%;
+      }
+      .column-md-8 {
+        width: 66.64%;
+      }
+      .column-md-10 {
+        width: 83.33%;
+      }
+      .time, .main {
+        font-size:28px;
+      }
+      ._margin-bottom-30 {
+        margin-bottom: 30px;
+      }
+      @media (max-width: 768px){
+        .column-xs-12 {
+          width: 100%;
+        }
+      }
+    `}</style>
+  </div>
+)
+
+const BreakRow = ({ time }) => (
+  <div className='row _margin-bottom-30'>
+    <div className='column-md-2 column-xs-12 time'>{time}</div>
+    <div className='column-md-8 column-xs-12 container-break'><div className='break' /></div>
+    <style jsx>{`
+      .row {
+        position: relative;
+        display: flex;
+        flex-wrap: wrap;
+        padding-left:7.5px;
+        padding-right:7.5px;
+      }
+      .column-md-5 {
+        width: 41.65%;
+      }
+      .column-md-2 {
+        width: 16.66%;
+      }
+      .column-md-8 {
+        width: 66.64%;
+      }
+      .column-md-10 {
+        width: 83.33%;
+      }
+      .time {
+        font-size:28px;
+      }
+      ._margin-bottom-30 {
+        margin-bottom: 30px;
+      }
+      .break {
+        height: 2px;
+        width: 20%;
+        background-color: #3A3A3A;
+      }
+      .container-break {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+      }
+      @media (max-width: 768px){
+        .container-break {
+          margin-top: 15px;
+        }
+        .column-xs-12 {
+          width: 100%;
+        }
+      }
+    `}</style>
+  </div>
+)
+
+const SessionRow = ({ time, children }) => (
+  <div className='row _margin-bottom-30'>
+    <div className='column-md-2 column-xs-12 time'>{time}</div>
+    {children}
+    <style jsx>{`
+      .row {
+        position: relative;
+        display: flex;
+        flex-wrap: wrap;
+        padding-left:7.5px;
+        padding-right:7.5px;
+      }
```

Comment:
Now, there are similar code in these lines:

- 99, 138, 190
- 103\~109, 142\~148, 245\~251

This “row” component can be extracted into `<ScheduleRow>`. The schedule row can have `margin-bottom: 30px` built in.

There are other similar code: `.column-md-5`, `.column-md-2`, `.column-md-8`, `.column-md-10`. They can be extracted into `<ScheduleColumn size={…}>`. Now responsive behavior can be controlled in one place, like this:

```js
function ScheduleColumn ({ children, size }) {
  return (
    <div className='column' data-size={size}>
      {children}
      <style jsx>{`
        .column { width: 100%; }
        @media (min-width: 768px) {
          .column[data-size="2"] { width: 16.66%; }
          .column[data-size="5"] { width: 41.65%; }
          .column[data-size="10"] { width: 83.33%; }
          .column[data-size="12"] { width: 83.33%; }
        }
      `}</style>
    </div>
  )
}
```

Actually we only need sizes 2, 5, 8, 10 to form a schedule layout.